src/OrbitGl/App.cpp

```diff
@@ -1147,166 +1159,145 @@ ErrorMessageOr<std::filesystem::path> OrbitApp::FindSymbolsLocally(
+        FireRefreshCallbacks();
+        return outcome::success();
+      });
 }
 
-ErrorMessageOr<void> OrbitApp::GetFunctionInfosFromHashes(
-    const ModuleData* module, const std::vector<uint64_t>& function_hashes,
-    std::vector<const FunctionInfo*>* function_infos) {
-  CHECK(module != nullptr);
-  const ProcessData* process = GetTargetProcess();
-  if (process == nullptr) {
-    return ErrorMessage(absl::StrFormat(
-        "Unable to get function infos for module \"%s\", because no process is selected",
-        module->file_path()));
+OrbitApp::LoadPresetModuleResult OrbitApp::LoadPresetModule(
+    const std::string& module_path, const orbit_client_protos::PresetModule& preset_module) {
+  ModuleData* module_data = module_manager_->GetMutableModuleByPath(module_path);
+  if (module_data == nullptr) {
+    return LoadPresetModuleResult::kModuleNotFound;
   }
-  if (!process->IsModuleLoaded(module->file_path())) {
-    return ErrorMessage(absl::StrFormat(
-        R"(Unable to get function infos for module "%s", because the module is not loaded by process "%s")",
-        module->file_path(), process->name()));
+
+  if (!module_data->is_loaded()) {
+    const auto result = main_thread_executor_->WaitFor(LoadModule(module_data));
+    if (result != MainThreadExecutor::WaitResult::kCompleted) return LoadPresetModuleResult::kAbort;
   }
 
-  size_t count_missing = 0;
-  for (const uint64_t function_hash : function_hashes) {
-    const FunctionInfo* function = module->FindFunctionFromHash(function_hash);
-    if (function == nullptr) {
-      count_missing++;
+  for (const uint64_t function_hash : preset_module.function_hashes()) {
+    const orbit_client_protos::FunctionInfo* const function_info =
+        module_data->FindFunctionFromHash(function_hash);
+    if (function_info == nullptr) {
+      LOG("Could not find function hash 0x%x in module %s", function_hash,
+          module_data->file_path());
       continue;
     }
-    function_infos->push_back(function);
+    SelectFunction(*function_info);
   }
-  if (count_missing != 0) {
-    return ErrorMessage(absl::StrFormat("* %d function infos missing from module \"%s\"\n",
-                                        count_missing, module->file_path()));
+  LOG("Auto hooked functions in module \"%s\"", module_data->file_path());
+
+  for (const uint64_t function_hash : preset_module.frame_track_function_hashes()) {
+    const orbit_client_protos::FunctionInfo* const function_info =
+        module_data->FindFunctionFromHash(function_hash);
+    if (function_info == nullptr) {
+      LOG("Could not find function hash 0x%x in module %s", function_hash,
+          module_data->file_path());
+      continue;
+    }
+    EnableFrameTrack(*function_info);
   }
-  return outcome::success();
+  LOG("Added frame tracks in module \"%s\"", module_data->file_path());
+  return LoadPresetModuleResult::kSuccess;
 }
 
 ErrorMessageOr<void> OrbitApp::SelectFunctionsFromHashes(
```

Comment:
This function always returns `success` now. Maybe return type should be void. Maybe we should think about whether the information that only some functions from a preset could be hooked is valuable for the user (and display a warning dialog)
Same for EnableFrameTrackFromHashes